### PR TITLE
Update yield-effects to use eslint suggestions feature

### LINKED ISSUE #71: yield-effects autofix modifies code functionality violating eslint best practices
I just tried to add this plugin to our codebase and noticed that if with code like this:
```js
const requests = endpoints.map(endpoint => call(fetch, endpoint));
yield all(requests);
```
The autofix for the `yield-effects` rule will update code to be:
```js
const requests = endpoints.map(endpoint => yield call(fetch, endpoint));
yield all(requests);
```

There are two less than ideal things here, so I'll open a second issue for the second request, but the first is that the runtime functionality of the code is changed. The [eslint best practices for applying fixes](https://eslint.org/docs/developer-guide/working-with-rules#applying-fixes-1) say this:
> Avoid any fixes that could change the runtime behavior of code and cause it to stop working

As of [eslint 6.7](https://eslint.org/blog/2019/11/eslint-v6.7.0-released#suggestions-api) (released Nov 2019), there is a new [suggestions API](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions) that allows editors to provide an optional ability for users to apply fixes that could change the underlying code functionality.

### PR DESCRIPTION
**Pull request by wdoug**
This makes it so the eslint autofix won't change code functionality.
Fixes #71

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1779 @@
+{
```
Comment (wdoug) [inline package-lock.json:1]: I'm not sure why this file wasn't previously committed, but I figured it would be fine to include in this PR. 

From the [npm docs](https://docs.npmjs.com/configuring-npm/package-lock-json.html):
> This file is intended to be committed into source repositories

File: package.json [hunk 2]
```diff
@@ -32,11 +32,11 @@
     "start": "mocha --watch --recursive --colors"
   },
   "peerDependencies": {
-    "eslint": ">= 4.1.0",
+    "eslint": ">= 6.7.0",
```
Comment (wdoug) [inline package.json:35]: I suppose this should elicit a major version bump of this package. I wasn't sure if I should include the change in this file in this pull request or if that should just be a separate commit.

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 node_js:
-- 6.2.0
```
Comment (wdoug) [inline .travis.yml:3]: https://eslint.org/docs/user-guide/migrating-to-6.0.0#nodejs-6-is-no-longer-supported

**Comment by pke** [comment]
Thanks for providing this PR to solve some basic problems with the current architecture.
Maybe we should change the travis node config to always test `lts` and `node` versions, so we would not have to update for every new released node version?

File: .travis.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
-- 10
+- 8.10.0
```
Comment (pke) [inline .travis.yml:3]: maybe use `lts` here.

File: .travis.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+- 14
```
Comment (pke) [inline .travis.yml:4]: And maybe use just `node` here for the very latest (stable) release?`

See https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions

File: .travis.yml (hunk 4 above)
Comment (wdoug) [reply .travis.yml:3]: Hmm. Based on the docs is it `lts` or `lts/*`? I guess I'll try one and see what happens...

**Comment by wdoug** [comment]
> Do you think those changes would require a major or patch version change?
> I lean toward patch since it fixes a bug (that would produce incorrect coder after applying fixes).

I will defer to [this SemVer discussion comment](https://github.com/semver/semver/issues/502#issuecomment-466501843) on the impact of updating peer dependencies:

> In an npm module, any peer dep change that removes a previously valid version from being valid is a major/breaking change. Peer deps are part of the public API.

I think this is consistent with how all the most popular libraries handle peer dependencies (for example eslint).

There might be a follow-up question about whether the peer dependency change is actually required for this specific change or whether it will just not have that feature if you're using an earlier version of the library. I'm not sure, but the functionality change seems like a breaking change to me anyway.

**Comment by wdoug** [comment]
Now that I think about this, at the very least this PR should also upgrade some docs to say that it is not fixable anymore. I'll submit that shortly.

**Comment by wdoug** [comment]
Also, I think the reasoning for this change might also apply to the `no-yield-in-race` rule. Should I go ahead and update that for this PR as well?

File: package-lock.json (hunk 1 above)
Comment (pke) [reply package-lock.json:1]: That's fine. Just an oversight.

**Comment by pke** [comment]
> Also, I think the reasoning for this change might also apply to the `no-yield-in-race` rule. Should I go ahead and update that for this PR as well?

Do you think it could also produce wrong fixes?

**Comment by wdoug** [comment]
Hmm. I guess I'm not so sure about that one. Maybe it is best to just do this for now and we could have a follow-up discussion/PR for the `no-yield-in-race` rule if needed.

**Comment by wdoug** [comment]
Out of curiosity, are you waiting on something for a new release with this change?

**Comment by pke** [comment]
Time issues. I really need to setup auto deploy with gh actions here.

**Comment by pke** [comment]
@wdoug released Thanks for your help!

**Comment by wdoug** [comment]
Hey @pke, I saw the GitHub release for this, but [npm still has it at version `1.1.3`](https://www.npmjs.com/package/eslint-plugin-redux-saga), so I assume it just didn't get published there yet.

**Comment by pke** [comment]
@wdoug I thought I had setup an action for that? Hmmm released manually now.

**Comment by wdoug** [comment]
Thanks
